sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/planner/BeamRelDataTypeSystem.java

```diff
@@ -35,4 +35,9 @@ public int getMaxNumericScale() {
   public int getMaxNumericPrecision() {
     return 38;
   }
+
+  @Override
+  public boolean shouldConvertRaggedUnionTypesToVarying() {
+    return true;
```

Comment:
nit: we should add comments about why all the stuff in this class is set to these values

Reply:
Added 'operators that change string length should return varchar.'